### DISCUSSION TITLE
请问mvvm 社区框架中 使用 RequestMessage如何在sender中传值呢?

### DISCUSSION THREAD

**Discussion by m3325599** [BYJRK/DotNet-Discussions#100] [MVVM]
请问mvvm 社区框架中 使用 RequestMessage如何在sender中传值呢?

我这样注册一个requestMessage订阅
 WeakReferenceMessenger.Default.Register<mainViewModel, RequestMessage<string>>(this, (r, m) =>
            {
                m.Reply("done");
            });

这样触发订阅,可以获取到返回值
  var tt = WeakReferenceMessenger.Default.Send<RequestMessage<string>>();

请问,我如果我需要在send中传入值,然后在订阅中处理后返回,如何能做到?,谢谢

**Comment by m3325599** [comment]
找到方法了,建立一个消息类,继承requestMessage,在消息类里带参,请问如果用 单单用requestMessage 能实现传参吗? 谢谢

**Comment by CodingOctocat** [comment]
使用泛型版本：`RequestMessage<T>`
